oauthenticator/github.py

```diff
@@ -205,23 +226,32 @@ async def update_auth_model(self, auth_model):
         # - about /user/emails: https://docs.github.com/en/rest/reference/users#list-email-addresses-for-the-authenticated-user
         #
         # Note that the read:user scope does not imply the user:emails scope!
+        # Note that the retrieved scopes will be empty for GitHub Apps https://docs.github.com/en/developers/apps
         access_token = auth_model["auth_state"]["token_response"]["access_token"]
         token_type = auth_model["auth_state"]["token_response"]["token_type"]
-        granted_scopes = auth_model["auth_state"].get("scope", [])
+        granted_scopes = [
+            scope for scope in auth_model["auth_state"].get("scope", []) if scope
+        ]
         if not user_info["email"] and (
-            "user" in granted_scopes or "user:email" in granted_scopes
+            "user" in granted_scopes
+            or "user:email" in granted_scopes
+            or not granted_scopes
```

Comment:
Can you explain this change? If this is called with empty scopes won't it fail to get the user's emails?

Reply:
The retrieved scopes will be empty for GitHub Apps https://docs.github.com/en/developers/apps.
So to facilitate this behavior, empty scopes are also allowed. 
This is also why we need a change in error handling of the email fetch. This might cause an error, but we don't know upfront what is causing this.